master/buildbot/changes/gerritchangesource.py

```diff
@@ -298,17 +312,49 @@ def streamProcessStopped(self):
             self.startStreamProcess()
             self.streamProcessTimeout = self.STREAM_BACKOFF_MIN
 
+    def buildGerritCommand(self, *gerrit_args):
+        '''Get an ssh command list which invokes gerrit with the given args on the
+        remote host'''
+
+        cmd = [
+            "ssh",
+            "%s@%s" % (self.username, self.gerritserver),
+            "-p", str(self.gerritport)
+        ]
+
+        if self.identity_file is not None:
+            cmd += ["-i", self.identity_file]
+
+        cmd.append("gerrit")
+        cmd.extend(gerrit_args)
+        return cmd
+
     def startStreamProcess(self):
         if self.debug:
             log.msg("starting 'gerrit stream-events'")
+
+        cmd = self.buildGerritCommand("stream-events")
         self.lastStreamProcessStart = util.now()
-        uri = "%s@%s" % (self.username, self.gerritserver)
-        args = [uri, "-p", str(self.gerritport)]
-        if self.identity_file is not None:
-            args = args + ['-i', self.identity_file]
-        self.process = reactor.spawnProcess(
-            self.LocalPP(self), "ssh",
-            ["ssh"] + args + ["gerrit", "stream-events"], env=None)
+        self.process = reactor.spawnProcess(self.LocalPP(self), "ssh", cmd, env=None)
+
+    @defer.inlineCallbacks
+    def getFiles(self, change, patchset):
+        cmd = self.buildGerritCommand("query", str(change), "--format", "JSON",
+                                      "--files", "--patch-sets")
+
+        if self.debug:
+            log.msg("querying gerrit for changed files in change %s/%s: %s" %
+                    (change, patchset, cmd))
+
+        out = yield utils.getProcessOutput(cmd[0], cmd[1:], env=None)
+        out = out.splitlines()[0]
+        res = json.loads(bytes2unicode(out))
+
+        if res.get("rowCount") == 0:
```

Comment:
```
if not res.get(...):
```

unless, of course, you want to explicitly catch the case when `rowCount` is present and is equal to 0.

Reply:
That is unlikely. If the change exists `gerrit query` will always return two lines of json - one with the result and one with `{"type": "stats", "rowCount": 1, ...}`. If it didn'tt exist then the response would be a single `{"type": "stats", "rowCount": 0}`. I think it's sufficient.

Reply:
i'm sorry what is unlikely?

Reply:
It's just that if the first line of output is `{"type": "stats", "rowCount": 0}` then no results were found, i.e. the change doesn't exist, which is unlikely given that we're reacting to an event about an existing change.